linux/net/rina/pdu.c

```diff
@@ -287,3 +288,9 @@ int pdu_destroy(struct pdu * p)
         return 0;
 }
 EXPORT_SYMBOL(pdu_destroy);
+
+uint_t pdu_length(struct pdu * pdu) {
+	return buffer_length(pdu->buffer) +
+		serdes_pci_size(pci_type(pdu->pci));
+}
+EXPORT_SYMBOL(pdu_length);
```

Comment:
Could be, I had some problem with this is serdes.c... 

I'm waiting for @sandervrijders ACK about this, because now serdes_pci_size is hardcoded with some values which are passed in the DIF configuration(addresses length and so on).

The part which depends on the serializer is in fact handled by that serializer procedure, but this procedure has a pdu context (length of the pdu, not of it's data).